File: Go/find-common-characters.go
func commonChars(words []string) []string {
    var freqMin, freqWord [26]int
    for i := range freqMin {
        freqMin[i] = 1000 // just need to be larger than 100
    }
    for i := range words {
        for _, c := range []byte(words[i]) {
            freqWord[c-'a']++
        }
        for c := range freqMin {
            freqMin[c] = min(freqMin[c], freqWord[c])
            freqWord[c] = 0
        }
    }
    result := make([]string, 0)
    for c := range freqMin {
        for i := 0; i < freqMin[c]; i++ {
            result = append(result, string(c + 'a'))
        }
    }
    return result
}

File: C++/find-common-characters.cpp
class Solution {
public:
    vector<string> commonChars(vector<string>& words) {
        vector<string> result;
        int count[26][100] = {0};
        int n = words.size();
        for (int i = 0; i < n; i++) {
            for (const char c : words[i]) {
                count[c-'a'][i]++;
            }
        }
        for (int c = 0; c < 26; c++) {
            int common = 100;
            for (int i = 0; i < n; i++) {
                common = min(common, count[c][i]);
            }
            string s(1, c + 'a');
            result.resize(result.size() + common, s);
        }
        return result;
    }
};